File: framework/ios/Example/Example/OpenpathWrapper/Models/ProvisionResponse.swift
import Foundation

struct ProvisionResponse: Decodable {
    let userOpal: String

    let environment: Environment
    struct Environment: Decodable {
        let heliumEndpoint: String
        let opalEnv: String
        let opalRegion: String
    }

    typealias User = OpenpathUser
    let user: User

    let credential: Credential
    struct Credential: Decodable {
        let id: Int
        let opal: String

        let credentialType: CredentialType
        struct CredentialType: Decodable {
            let modelName: String
        }

        let mobile: Mobile
        struct Mobile: Decodable {
            let id: Int
            let name: String
            let provisionedAt: String
        }
    }
}

File: types/Events/ProvisionResponse.ts
import { User } from '../Models/User';

/**
 * The data included with the "onProvisionResponse" event.
 *
 * Invoked when {@link provision} is complete.
 *
 * @event onProvisionResponse
 * @see {@link OpenpathEvent}
 */
export interface ProvisionResponse {
  userOpal: string;

  environment: {
    heliumEndpoint: string;
    opalEnv: string;
    opalRegion: string;
  };

  user: User;

  credential: {
    id: number;
    opal: String;

    credentialType: {
      modelName: string;
    };

    mobile: {
      id: null;
      name: string;
      provisionedAt: string;
    };
  };
}
